### ISSUE TITLE
Error sharing Angular 13 in production

### ISSUE THREAD

**Issue by AlbaSS18** [single-spa/single-spa-angular#431]
### Question

Hello,

A week ago, I got to share Angular 13 thanks to the help provided in this issue https://github.com/single-spa/single-spa-angular/issues/418 . Everything were well in local. However, now I have deployed my app in production and this error appears.

![image](https://user-images.githubusercontent.com/46462037/162390317-88266abc-fac5-4abd-b784-a1193055c0a9.png)

In the root-config, I have `importmap.json` file. In this file, I set the next bundles:

```
"single-spa": "https://cdn.jsdelivr.net/npm/single-spa@5.9.2/lib/umd/single-spa.min.js",
"single-spa-angular/internals": "https://cdn.jsdelivr.net/npm/single-spa-angular@4.9.1/bundles/single-spa-angular-internals.umd.js",
"single-spa-angular": "https://cdn.jsdelivr.net/npm/single-spa-angular@4.9.1/bundles/single-spa-angular.umd.js",

"@angular/core": "https://unpkg.com/@angular/core@11.0.5/bundles/core.umd.min.js",
"@angular/compiler": "https://unpkg.com/@angular/compiler@11.0.5/bundles/compiler.umd.min.js",
"@angular/common": "https://unpkg.com/@angular/common@11.0.5/bundles/common.umd.min.js",
"@angular/common/http": "https://unpkg.com/@angular/common@11.0.5/bundles/common-http.umd.min.js",

"@angular/platform-browser-dynamic": "https://unpkg.com/@angular/platform-browser-dynamic@11.0.5/bundles/platform-browser-dynamic.umd.min.js",
"@angular/platform-browser": "https://unpkg.com/@angular/platform-browser@11.0.5/bundles/platform-browser.umd.min.js",
"@angular/platform-browser/animations": "https://unpkg.com/@angular/platform-browser@11.0.5/bundles/platform-browser-animations.umd.min.js",

"@angular/forms": "https://unpkg.com/@angular/forms@11.0.5/bundles/forms.umd.min.js",
"@angular/router": "https://unpkg.com/@angular/router@11.0.5/bundles/router.umd.min.js",
"@angular/animations": "https://unpkg.com/@angular/animations@11.0.5/bundles/animations.umd.min.js",
"@angular/animations/browser": "https://unpkg.com/@angular/animations@11.0.5/bundles/animations-browser.umd.min.js",

...
```
Then, in the app, I have the `extra-webpack.config.js` and share Angular like this:
```
module.exports = (angularWebpackConfig, options) => {
  const singleSpaWebpackConfig = singleSpaAngularWebpack(angularWebpackConfig, options);

  singleSpaWebpackConfig.externals = [
    /^@monitor\/*/,
    /^single-spa$/,
    /^single-spa-angular$/,

    /^rxjs$/,
    /^rxjs\/operators$/,

    // angular
    /^@angular\/core$/,
    /^@angular\/compiler$/,
    /^@angular\/common$/,
    /^@angular\/common\/http$/,
    /^@angular\/forms$/,
    /^@angular\/router$/,
    /^@angular\/animations$/,
    /^@angular\/animations\/browser/,
    /^@angular\/platform-browser$/,
    /^@angular\/platform-browser-dynamic$/,
    /^@angular\/platform-browser\/animations$/,
    /^zone\.js$/,

   /^i18next$/,
  ];

  return singleSpaWebpackConfig
}
```

I have been searching in other issues but I don't find anything about this error. Do I need to configure something to make it work in production?

Thank you in advance :smiley: 

### Environment

<pre><code>
Libs:
- @angular/core version: 13.3.1
- single-spa-angular version: 6.3.0

**Comment by arturovt** [comment]
The `CompilerFactory` is a JIT-only class, which means it should never be shipped into production. It's provided by the `platformBrowserDynamic`, which is replaced with `platformBrowser` when the app is built in the production mode. It still calls `bootstrapModule` in the call stack which then calls `compileNgModuleFactory`, which should not actually do any compilation in the production mode.

The second question is you're saying you're sharing Angular 13 dependencies, but I see Angular 11 version in the root-config import map.

**Comment by arturovt** [comment]
Closing the issue due to the absence of the reproducible example. Feel free to re-open when one is provided.

**Comment by AlbaSS18** [comment]
Hello,

Sorry for the delay in replying. These days, I have been on holidays. 

First, I have edited the question because I copied Angular dependencies wrong in the comment. 

Second, I think I have solved the problem. We use [import-map-overrides](https://github.com/joeldenning/import-map-overrides) with some dependencies overwritten by default (@angular/router, @angular/core, @angular/common... between them) and the dependencies' url wasn't updating. They pointed at Angular 11. Reset all overrides solved the problem, I think. However, I will carry on trying Angular 13. 

Thank you very much for your quick answer. Stay safe :heart:
